pkg/service/managedautoevents.go

```diff
@@ -35,35 +36,43 @@ func (s *DeviceService) AddDeviceAutoEvent(deviceName string, event contract.Aut
 	}
 
 	if !found {
-		s.LoggingClient.Debug(fmt.Sprintf("Adding new auto event to device %s: %v\n", deviceName, event))
+		s.LoggingClient.Debugf("Adding new AutoEvent %s to device %s", event.Resource, deviceName)
 		device.AutoEvents = append(device.AutoEvents, event)
-		cache.Devices().Update(device)
+		err := cache.Devices().Update(device)
+		if err != nil {
+			s.LoggingClient.Errorf("failed to update device %s with AutoEvent change", deviceName)
+			return err
+		}
 	}
-
-	autoevent.GetManager().RestartForDevice(deviceName, nil)
+	s.manager.RestartForDevice(deviceName)
 
 	return nil
 }
 
 // RemoveDeviceAutoEvent removes an AutoEvent from the Device with given name
-func (s *DeviceService) RemoveDeviceAutoEvent(deviceName string, event contract.AutoEvent) error {
+func (s *DeviceService) RemoveDeviceAutoEvent(deviceName string, event models.AutoEvent) error {
 	device, ok := cache.Devices().ForName(deviceName)
 	if !ok {
-		msg := fmt.Sprintf("Device %s cannot be found in cache", deviceName)
+		msg := fmt.Sprintf("failed to find device %s cannot in cache", deviceName)
 		s.LoggingClient.Error(msg)
 		return fmt.Errorf(msg)
 	}
 
-	autoevent.GetManager().StopForDevice(deviceName)
+	s.manager.StopForDevice(deviceName)
```

Comment:
do we need to stop it here?
it is restarted in the end of the function.

Reply:
you're right we don't need to stop here.